File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/models/consumer/datasync/entity/PNDataSyncGetEntitiesResult.java
package com.pubnub.api.java.models.consumer.datasync.entity;

import com.pubnub.api.java.models.consumer.datasync.PNDataSyncPage;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.jetbrains.annotations.NotNull;

import java.util.List;

/**
 * Result of {@code pubnub.dataSync().getEntities()}.
 * DataSync list endpoints are cursor-based.
 */
@Getter
@NoArgsConstructor
@ToString
public class PNDataSyncGetEntitiesResult {
    private int status;
    private List<PNDataSyncEntity> data;

    /**
     * Cursor-based pagination metadata. Always present (an empty page when the server sends no
     * {@code meta}); read {@link PNDataSyncPage#isHasNext()} to detect the last page.
     */
    @NotNull
    private PNDataSyncPage next;

    public PNDataSyncGetEntitiesResult(int status, List<PNDataSyncEntity> data, @NotNull PNDataSyncPage next) {
        this.status = status;
        this.data = data;
        this.next = next;
    }
}

File: pubnub-kotlin/pubnub-kotlin-api/src/commonMain/kotlin/com/pubnub/api/models/consumer/datasync/entity/PNDataSyncGetEntitiesResult.kt
package com.pubnub.api.models.consumer.datasync.entity

import com.pubnub.api.models.consumer.datasync.PNDataSyncPage

/**
 * Result of `pubnub.dataSync.getEntities`.
 *
 * DataSync list endpoints are cursor-based.
 *
 * @property status HTTP status code of the response.
 * @property data The entities on this page.
 * @property next Cursor-based pagination metadata. Always present (an empty page when the server sends
 *   no `meta`); read [PNDataSyncPage.hasNext] to detect the last page, never `next == null`.
 */
data class PNDataSyncGetEntitiesResult(
    val status: Int,
    val data: List<PNDataSyncEntity>,
    val next: PNDataSyncPage,
)
